src/assets/src/components/common.tsx

```diff
@@ -260,27 +260,22 @@ export const EditToggleField: React.FC<EditToggleFieldProps> = (props) => {
     return contents;
 }
 
-export interface DropdownValue {
-    value: string;
-    displayValue: string;
-}
-
 interface BackendSelectorProps {
-    options: DropdownValue[];
+    allowedBackends: Set<string>;
+    backends: {[backend_type: string]: string};
```

Comment:
1. The previous factoring necessitated the creation of an extra interface, which can now be removed
2. Previously, logic transforming `(allowedBackends, backends) => DropdownValue[]` was duplicated in two spots (the two spots where BackendSelector is used). The duplication has been removed
3. Since this is a BackendSelector, and not just a generic Selector, it seems appropriate that it would handle some work specific to backends, instead of taking a nondescript list of "options"

WRT the use of `Set`, I use it to indicate to the reader that a list of allowedBackends should logically have no duplicates. Functionally, it makes it impossible to pass in a list with duplicates.

Reply:
Yeah, I'm still learning about where the backend data comes from. Looks like it's important to differentiate between `possibleBackends` (now just `backends` from the globals) and `allowedBackends`. Just wasn't sure if this component needs to know that distinction. These props get passed through a whole lot of components, but I guess that's better than duplicating the logic in a couple other higher-order components.

Reply:
So `Set` is mostly for readability. Just making sure I wasn't missing something.